lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -728,6 +729,22 @@ LogicalResult ReshapeOp::verify() {
 }
 
 //-- FpToFpOp --
+OpFoldResult FpToFpOp::fold(FoldAdaptor adaptor) {
+  auto srcVal = getSrc();
+  if (!isZeroConst(srcVal)) {
+    return {};
+  }
+
+  auto srcConstant = srcVal.getDefiningOp<arith::ConstantOp>();
```

Comment:
We need to be a bit careful in general regarding floating point calcuations. I think folding zero converions are generally okay; but still we may want to be a bit clear regarding +/-0 for example:

```c++
if (matchPattern(getSrc(), m_PosZeroFloat()) {
  return // postive zero attr for dstTy
}
if (matchPattern(getSrc(), m_NegZeroFloat()) {
  return // negative zero attr for dstTy
}